### ISSUE TITLE
Not translated text in column menu

### ISSUE THREAD

**Issue by jens-maus** [jens-maus/yam#541]
**Originally by _s.hawamdeh@teletu.it_ on 2014-03-22 13:35:07 +0100**

___
The 4 options in the "Column" menu of YAM is translable (they are all availible in the original catalog descriptor) but they doesn't work, infact even if you use another language that 4 strings still in english

The "broken" strings are:

msgctxt "MSG_MA_CTX_DEFWIDTH_THIS (1476//)"
msgid "Default width: this"

msgctxt "MSG_MA_CTX_DEFWIDTH_ALL (1477//)"
msgid "Default width: all"

msgctxt "MSG_MA_CTX_DEFORDER_THIS (1478//)"
msgid "Default order: this"

msgctxt "MSG_MA_CTX_DEFORDER_ALL (1479//)"
msgid "Default order: all"

**Comment by tboeckel** [comment]
**Originally on 2014-03-22 14:41:29 +0100**

___
I suppose you are refering to the context menus which are created for the title buttons of the folder listtree and mail list, correct?

For me the 4 items perfectly appear in german language, and the strings definitely stem from YAM's catalog, not the one of the NList classes. It can also easily be seen in the source code that the strings are definitely used, i.e. in mui/MainFolderListtree.c lines 529 to 532:

```
        Child, MenuitemObject, MUIA_Menuitem_Title, tr(MSG_MA_CTX_DEFWIDTH_THIS), MUIA_Menuitem_CopyStrings, FALSE, MUIA_UserData, MUIV_NList_Menu_DefWidth_This, End,
        Child, MenuitemObject, MUIA_Menuitem_Title, tr(MSG_MA_CTX_DEFWIDTH_ALL),  MUIA_Menuitem_CopyStrings, FALSE, MUIA_UserData, MUIV_NList_Menu_DefWidth_All,  End,
        Child, MenuitemObject, MUIA_Menuitem_Title, tr(MSG_MA_CTX_DEFORDER_THIS), MUIA_Menuitem_CopyStrings, FALSE, MUIA_UserData, MUIV_NList_Menu_DefOrder_This, End,
        Child, MenuitemObject, MUIA_Menuitem_Title, tr(MSG_MA_CTX_DEFORDER_ALL),  MUIA_Menuitem_CopyStrings, FALSE, MUIA_UserData, MUIV_NList_Menu_DefOrder_All,  End,
```


Please keep in mind that just copying a new version of a catalog it over the previous version is not enough if the program using that catalog has been run before. Like libraries catalogs are kept in memory as long as there is enough free memory available. To enforce the usage of the new catalog you have to flush the old version from memory first. Flushing all catalogs from memory can be done by calling "expunge locale.library" on AmigaOS4 and "avail flush" on all other systems.

**Comment by jens-maus** [comment]
**Originally by _s.hawamdeh@teletu.it_ on 2014-03-23 13:32:08 +0100**

___
@Thore

> I suppose you are refering to the context menus which are created for the title
> buttons of the folder listtree and mail list, correct?

I mean that area --> http://s7.postimg.org/3u6k9fg4r/yam.png
Are we referring to the same area ?

> Please keep in mind that just copying a new version of a catalog it over the > previous version is not enough if the program using that catalog has been run > before. Like libraries catalogs are kept in memory as long as there is enough > free memory available.

I don't think it's that because if i remember right i always had this specific problem on that menu (sorry if i report it only now), aniway i follow your suggestion expunging the library, then i did an hard reboot, as a test i also localized my system in order to use German language only, but this confirm my issue because my entire system is now in German but that column menu in YAM still always in English

**Comment by tboeckel** [comment]
**Originally on 2014-03-23 19:05:12 +0100**

___
Ok, now I know what you are talking about. You definitely omitted the very important information that you are opening the context menu for an empty folder. In this case you really get the context menu created by NList.mcc instead of the one created by YAM. Unfortunately the NList menu item texts are static and cannot be translated (yet). I will see what I can do about that.

**Comment by tboeckel** [comment]
**Originally on 2014-03-23 19:23:51 +0100**

___
In (cc3706d):
* mui/MainMailList.c: changed the (yet) non-translatable context menu of NList.mcc for empty folders by the same context menu as is created for the list title bar. This closes #541.
